network/src/router.rs

```diff
@@ -46,24 +53,86 @@ fn attestation_topic(parent_hash: Hash) -> Hash {
 	BlakeTwo256::hash(&v[..])
 }
 
+fn incoming_message_topic(parent_hash: Hash, parachain: ParaId) -> Hash {
+	let mut v = parent_hash.as_ref().to_vec();
+	parachain.using_encoded(|s| v.extend(s));
+	v.extend(b"incoming");
+
+	BlakeTwo256::hash(&v[..])
+}
+
+/// Receiver for block data.
+pub struct BlockDataReceiver {
+	outer: ::futures::sync::oneshot::Receiver<::futures::sync::oneshot::Receiver<BlockData>>,
+	inner: Option<::futures::sync::oneshot::Receiver<BlockData>>
```

Comment:
Comments here on what exactly these fields are would be helpful.

Reply:
(gregory has added these in the last PR, I just moved it around)

They'll probably disappear soon when we get a smarter gossip protocol.